code/game/atoms_movable.dm

```diff
@@ -632,7 +632,7 @@
 	var/area/oldarea = get_area(oldloc)
 	var/area/newarea = get_area(newloc)
 
-	loc = newloc
+	change_loc(newloc)
```

Comment:
this is dumb why do we need yet another wrapper

Reply:
normal movement:
Move(B) -> loc = B -> A.Exited() & B.Entered() -> Moved(old_loc = A)
												^ here be side effects 		   ^ tell everything that cares that movement from A->B is done
movement where the mover gets deleted in Entered() (e.g. this problem):
![image](https://user-images.githubusercontent.com/15794172/237006746-2542a54c-7e82-4ac5-bdfd-82a55d25e17e.png)

for this problem, i need to be able to update grid loc before side effects from Entered()/Exited() happen, because those side effects can be a second movement that finishes before the first movement does, or they can delete the mover which itself involves a nested movement to nullspace. i cant move Entered and Exited to after Moved() though, because the only thing there that actually relies on Moved being after those two is connect_loc (otherwise it will react to tracked moving into each tile). 

however, i also just found out why stuff is getting stuck in the grid in the first place, so i need to redo this anyways

Reply:
right, I guess i'm just concerned we end up with an endless set of wrappers as things "care" about location changes before movement effects.

I guess if we can contain it all to this one proc it will be fine.